client/src/ui/EventListeners.res

```diff
@@ -0,0 +1,105 @@
+let decodeTransEvent = (fn: string => 'a, j): 'a => {
+  open Json.Decode
+  fn(field("propertyName", string, j))
+}
+
+let decodeAnimEvent = (fn: string => 'a, j): 'a => {
+  open Json.Decode
+  fn(field("animationName", string, j))
+}
+
+// Generic event, the the listener handle and do what it wants with the event object
+let onEvent = (
+  ~event: string,
+  ~key: string,
+  ~preventDefault=true,
+  listener: Web.Node.event => 'msg,
+): Vdom.property<'msg> =>
+  Tea.Html.Events.onCB(event, key, evt => {
+    if preventDefault {
+      evt["preventDefault"]()
+    }
+    Some(listener(evt))
+  })
+
+let eventBoth = (~key: string, event: string, constructor: MouseEvent.t => 'msg): Vdom.property<
+  'msg,
+> =>
+  Tea.Html.Events.onWithOptions(
+    ~key,
+    event,
+    {stopPropagation: false, preventDefault: false},
+    Decoders.wrapDecoder(Json.Decode.map(constructor, MouseEvent.decode)),
+  )
```

Comment:
What does this do?

Reply:
I'm not sure what the 'both' and 'neither' (below) are referring to.

Reply:
Both and neither refer to stopPropagation and preventDefault